docs/reference/indices/create-index.asciidoc

```diff
@@ -44,6 +44,7 @@ Index names must meet the following criteria:
 - Cannot start with `-`, `_`, `+`
 - Cannot be `.` or `..`
 - Cannot be longer than 255 bytes (note it is bytes, so multi-byte characters will count towards the 255 limit faster)
+- Names starting with `.` are deprecated as these names are reserved for internal indices managed by plugins
```

Comment:
I'm not sure we can just drop it, because per the most recently discussed design, `.`-prefixed indices still must be either hidden or system, or else they'll get a deprecation warning. I did update the note, though.